### PR TITLE
wpsoffice-cn-365: init at 12.8.2.16969

### PR DESCRIPTION
**Pull request by pokon548**
## Description of changes
Introduce `wpsoffice-cn-365`, another WPS Office variant. It includes a bunch of features that has not included in `wpsoffice-cn`. 
Website: https://365.wps.cn/home

### Known issues
- ~~Integrated mode is not working. Switching to this window management mode will render `wpsoffice-cn-365` not able to launch.~~ Fixed.
- ~~Login status will not survive upon each launch.~~ Fixed.
- ~~CEF based apps cannot use input methods (tested using `fcitx5`).  ~~Possibly upstream issue.~~ Caused by built-in `libstdc++.so.6` not able to find libraries: `libstdc++.so.6: version GLIBCXX_3.4.29' not found (required by /nix/store/xxxxx-fcitx5-with-addons-5.1.9/lib/gtk-3.0/3.0.0/immodules/im-fcitx5.so)`. Feel free to make pr if you find any solutions.~~ Fixed.
	- Qt based apps (Writer, Table, Presentation, etc.) is *not* affected (when using `QT_IM_MODULE=fcitx5`. `QT_IM_MODULE=wayland` is not supported. `QT_IM_MODULE=ibus` is untested).

### Notes
- **Will periodically read system info (even if you choose to disable cloud abilities in `oem.ini`)**, including (but possibly not limited to):
	- `/usr/local/var/lib/dbus/machine-id` (Not existed in NixOS)
	- `/proc/sys/kernel/random/boot_id`
	- `/var/lib/dbus/machine-id` (Not existed in NixOS)
- Requires software activation.
	- `misc-linux` will write activation info into `/opt/kingsoft`. Make sure you give write permission to this path, otherwise activation will not work even if you have a legit license.
	- You cannot use a personal VIP account to activate `wpsoffice-cn-365`.
	- If you choose not to activate, then you must log in to use `wpsoffice-cn-365` after the 30 days trial is over.

### Screenshots
截图 2024-06-09 22-05-50
截图 2024-06-09 22-07-29

~~Currently, this pr is based upon https://github.com/NixOS/nixpkgs/pull/318500 and will be rebased once the prior one is merged.~~ Merged.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [NixOS/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wineee** [comment]
It‘s better to add an separate package for wpsoffice-cn-365, it is too complicated to have three versions in the wps package at the same time

File: pkgs/applications/office/wpsoffice/default.nix [hunk 1]
```diff
@@ -100,6 +145,11 @@ stdenv.mkDerivation rec {
     "libQtCore.so.4"
     "libQtNetwork.so.4"
     "libQtXml.so.4"
+    # WPS 365 specific: file manager integration. Not needed
+    "libnautilus-extension.so.1"
+    "libcaja-extension.so.1"
+    # WPS 365 specific: ukylin specific lib. Not needed on NixOS
```
Comment (wineee) [inline pkgs/applications/office/wpsoffice/default.nix:151]:
```suggestion
```
peony is also a file manager like nautilus

**Comment by pokon548** [comment]
> It‘s better to add an separate package for wpsoffice-cn-365, it is too complicated to have three versions in the wps package at the same time

For me, it is fine to keep `wpsoffice-365` at the same package, as most of the changes (dependencies, binary patches, etc.) are expected to be shareable as the personal variant reaches `12` version.

File: pkgs/applications/office/wpsoffice/default.nix (hunk 1 above)
Comment (pokon548) [reply pkgs/applications/office/wpsoffice/default.nix:151]: > peony is also a file manager like nautilus

I see. However, from my experience, file manager related libraries are only used to inject some entries (like cloud drive, etc.) into file manager, which should not breaks major usability. Maybe we could leave this as-is until it is indeed important.

**Comment by pokon548** [comment]
Replaced by https://github.com/NixOS/nixpkgs/pull/329296

**Pull request closed without merging**